File: clojure/factorial.clj
(ns clojure.misc.factorial
	(:gen-class))

(defn factorial [num]
  (cond 
    (= num 0) 1 ;Factorial of 0 equals 1.
    :else (* num (factorial (- num 1)))
  )
)

(defn main []
  (def nums (list 9 6 3))
  (doseq [num nums] (println (factorial num)))
)

(main)

File: elixir/factorial.ex
defmodule Factorial do
  def start do
    nums = [9, 3, 6]
    for num <- nums do
      IO.puts("#{num}! is #{factorial(num)}")
    end
  end

  def factorial(0), do: 1 # Factorial of 0 equals 1.
  def factorial(num) do
    num * factorial(num - 1)
  end
end
